packages/flutter_tools/lib/src/commands/migrate.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// import 'package:process/process.dart';
+
+// import '../base/file_system.dart';
+import '../base/logger.dart';
+// import '../base/platform.dart';
+import '../base/terminal.dart';
+import '../migrate/migrate_utils.dart';
+import '../runner/flutter_command.dart';
+// TODO(garyq): Add each of these back in as they land.
+// import 'migrate_abandon.dart';
+// import 'migrate_apply.dart';
+// import 'migrate_resolve_conflicts.dart';
+// import 'migrate_start.dart';
+// import 'migrate_status.dart';
+
+/// Base command for the migration tool.
+class MigrateCommand extends FlutterCommand {
+  MigrateCommand({
+    // bool verbose = false,
+    required this.logger,
+    // TODO(garyq): Add each of these back in as they land.
+    // required FileSystem fileSystem,
+    // required Terminal terminal,
+    // required Platform platform,
+    // required ProcessManager processManager,
+  }) {
+    // TODO(garyq): Add each of these back in as they land.
+    // addSubcommand(MigrateAbandonCommand(logger: logger, fileSystem: fileSystem, terminal: terminal, platform: platform, processManager: processManager));
+    // addSubcommand(MigrateApplyCommand(verbose: verbose, logger: logger, fileSystem: fileSystem, terminal: terminal, platform: platform, processManager: processManager));
+    // addSubcommand(MigrateResolveConflictsCommand(logger: logger, fileSystem: fileSystem, terminal: terminal));
+    // addSubcommand(MigrateStartCommand(verbose: verbose, logger: logger, fileSystem: fileSystem, platform: platform, processManager: processManager));
+    // addSubcommand(MigrateStatusCommand(verbose: verbose, logger: logger, fileSystem: fileSystem, platform: platform, processManager: processManager));
+  }
+
+  final Logger logger;
+
+  @override
+  final String name = 'migrate';
+
+  @override
+  final String description = 'Migrates flutter generated project files to the current flutter version';
+
+  @override
+  String get category => FlutterCommandCategory.project;
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{};
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    return const FlutterCommandResult(ExitStatus.fail);
+  }
+}
+
+Future<bool> gitRepoExists(String projectDirectory, Logger logger, MigrateUtils migrateUtils) async {
```

Comment:
Is this called somewhere? If not, it should be held back and only checked in once it's used/tested somewhere.